File: redsheet-portal/src/app/database-models/v-suggested-action.ts
import { required, maxLength, range } from '@rx/annotations';

export class vSuggestedAction {
    constructor(vSuggestedAction?: vSuggestedAction )  {
        let properties = [ "cultureCountryId", "suggestedActionId", "suggestedActionValue",];
        for (let property of properties)
            if (vSuggestedAction && vSuggestedAction[property])
                this[property] = vSuggestedAction[property];
    }
 
	cultureCountryId : number =   0 ;
 
    @range(1,2147483647)
	suggestedActionId : number =   undefined;
 
    @required()
    @maxLength(500)
	suggestedActionValue : string =   undefined;
}

File: RedSheet.DbEntities/Models/vSuggestedAction.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vSuggestedActions", Schema = "dbo")]
	[Serializable]
	public partial class vSuggestedAction  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int CultureCountryId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int SuggestedActionId { get; set; } 
 
	[Required] 
	[MaxLength(500)]
	public string SuggestedActionValue { get; set; } 

	public vSuggestedAction()
	{
    }

    }
}

